src/axom/sidre/core/Group.cpp

```diff
@@ -1790,6 +1790,31 @@ bool Group::isEquivalentTo(const Group* other, bool checkName) const
   return is_equiv;
 }
 
+
+void Group::checkConduitCall(const std::function<void(void)> & conduitOp) const
+{
```

Comment:
Do we want to use a `std::mutex` here?

Reply:
Is it cleaner to handle in Conduit?

Reply:
It might be marginally cleaner to Axom, a user code, but each of Conduit's user codes will need to handle errors slightly differently.  Having the "save handlers, do operation, restore handlers" logic here, as we previously agreed, is a little bit of awkwardness---but it is appropriate to Axom, and we can change or fix it easier than if the code were in Conduit or any other library.  I don't see a good reason to put this kind of a `checkConduitCall` into Conduit, unless @cyrush already intended to do this.  @cyrush , do you have anything to add?

Reply:
I am confused, I don't think the group api implementation is setup to support multi-threaded access or modification? 

Also, HDF5 will not work with multi-threading unless built with special options and even with those settings I think there are some limitations

Reply:
@agcapps based on the discussion here, did you decide whether a mutex is needed?